core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1535,16 +1524,26 @@ public void run() {
                   ChannelLogLevel.INFO,
                   "Config selector from name resolver discarded by channel settings");
             }
-            configSelector.set(null);
+            configSelector.set(effectiveServiceConfig.getDefaultConfigSelector());
           } else {
             // Try to use config if returned from name resolver
             // Otherwise, try to use the default config if available
             if (validServiceConfig != null) {
               effectiveServiceConfig = validServiceConfig;
-              configSelector.set(resolvedConfigSelector);
+              if (resolvedConfigSelector != null) {
+                configSelector.set(resolvedConfigSelector);
+                if (effectiveServiceConfig.getDefaultConfigSelector() != null) {
+                  channelLogger.log(
+                      ChannelLogLevel.INFO,
+                      "Method configs in service config will be discarded due to presence of"
```

Comment:
I'm not wild about this channel logging, as it seems like spam and is unlikely to help notice a bug. But we can leave it for now and remove it if it becomes a problem.

Reply:
With the new implementation, the converted config selector will never be null. Removed this block.